doc/nrf/app_memory.rst

```diff
@@ -67,6 +67,65 @@ Besides applying `General recommendations`_, you can also complete the following
 
 For reference, you can find minimal footprint configurations of the :ref:`peripheral_lbs` sample in :file:`nrf/samples/bluetooth/peripheral_lbs/minimal.conf` and the :ref:`peripheral_uart` sample in :file:`nrf/samples/bluetooth/peripheral_uart/minimal.conf`.
 
+.. _app_memory_bt_mesh:
+
+Bluetooth mesh
+**************
+
+Besides applying `General recommendations`_ and Bluetooth_ optimization actions, you can use the following configuration options to optimize the :ref:`Bluetooth mesh <ug_bt_mesh>` part of your application:
```

Comment:
Sorry, but I find this quite confusing from the point of view of a casual SDK user. The page's introduction starts with a talk about RAM and ROM optimization (I would even object to the use of the technically incorrect word - ROM, it should be FLASH, but lets keep it aside for a moment). 

Here several options are presented without telling a consequence of what happens with RAM or Flash or both when these options are used. Without this context, this is not useful and leads to confusion. Several sections on this page have the same problem, it is just a collection of bunch of options without clear indication on which option changes what footprint. It is left to the users to let them dig-down and find out if they can understand that.

We can at least streamline this for mesh section with small tweaks to the paragraph sentences or adding tables.

The section should say something like this (I have simplified the formatting for illustration, you can format it as needed), or you can skip the table and spell it out in words in sentences.

```
RAM footprint and the amount of flash required for persistent storage <insert correct term here, partition?> is affected by the choice of values for these options. Higher the values, more resources would be used:

Option | RAM impact | Persistent storage impact |
========================================
:kconfig:option:`BT_MESH_MODEL_KEY_COUNT` | y | y |
:kconfig:option:`BT_MESH_SUBNET_COUNT` | y | y |
:kconfig:option:`BT_MESH_APP_KEY_COUNT` | y | y |
:kconfig:option:`BT_MESH_LABEL_COUNT` | y | y |

 
* Configuration options for segmented messages (for example how many segmented messages a node can send or receive, and the number of segments per message):

Option | RAM impact | Persistent storage impact |
========================================
 :kconfig:option:`BT_MESH_TX_SEG_MSG_COUNT` | y | n |
 :kconfig:option:`BT_MESH_RX_SEG_MSG_COUNT` | y | n |
 :kconfig:option:`BT_MESH_SEG_BUFS` | y | n |
 :kconfig:option:`BT_MESH_RX_SEG_MAX` | y | n |
 :kconfig:option:`BT_MESH_TX_SEG_MAX` | y | n |

* General node configuration:


Option | RAM impact | Persistent storage impact |
========================================
:kconfig:option:`BT_MESH_MODEL_KEY_COUNT` | y | y |
:kconfig:option:`BT_MESH_MODEL_GROUP_COUNT` | y | y |
:kconfig:option:`BT_MESH_SUBNET_COUNT` | y | y |
:kconfig:option:`BT_MESH_APP_KEY_COUNT` | y | y |
:kconfig:option:`BT_MESH_LABEL_COUNT` | y | y |

... 

* Advertiser/Extended advertising/Legacy advertiser/Friend feature/ Proxy/ Other device/ configuration:

.... | y | n | (for all)

* Provisioner
.... | y | y | (for all)


```

Reply:
@omkar3141 , please have a look at NCSDK-17146. I guess this task covers what you are suggesting (and a bit more).

Reply:
Instead of doing a table with | y | y/n |, we could just say in the head line that this effects (RAM) or (RAM+ROM). That way it is not a big table. They are the same within each section. (ROM->FLASH if that is changed in the page.)

Reply:
This is exactly what I did in my proposal above.